src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/button_press_request.cc

```diff
@@ -1,20 +1,232 @@
+    return false;
+  }
+
+  if (enums_value::kRadio == module_type) {
+    if (!helpers::in_range(buttons_radio(), button_name)) {
+      LOG4CXX_WARN(logger_,
+                   "Trying to acceess climate button with module type radio");
+      return false;
+    }
+  }
+
+  if (enums_value::kClimate == module_type) {
+    if (!helpers::in_range(buttons_climate(), button_name)) {
+      LOG4CXX_WARN(logger_,
+                   "Trying to acceess radio button with module type climate");
+      return false;
+    }
+  }
   return true;
 }
-void ButtonPressRequest::Run() {}
-void ButtonPressRequest::on_event(
-    const application_manager::event_engine::Event& event) {}
+
+void ButtonPressRequest::Execute() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const std::string button_name =
+      (*message_)[strings::msg_params][message_params::kButtonName].asString();
+  const std::string module_type =
+      (*message_)[strings::msg_params][message_params::kModuleType].asString();
+
+  static ButtonsMap btn_map = buttons_map();
+  mobile_apis::ButtonName::eType button_id =
+      mobile_apis::ButtonName::INVALID_ENUM;
+  if (btn_map.end() != btn_map.find(button_name)) {
+    button_id = btn_map[button_name];
+  }
+
+  const smart_objects::SmartObject* rc_capabilities =
+      application_manager_.hmi_capabilities().rc_capability();
+  const bool button_name_matches_module_type =
+      CheckButtonName(module_type, button_name, rc_capabilities);
+  const bool button_id_exist_in_caps =
+      rc_capabilities &&
+      CheckIfButtonExistInRCCaps(*rc_capabilities, button_id);
+
+  if (button_name_matches_module_type && button_id_exist_in_caps) {
+    SendHMIRequest(hmi_apis::FunctionID::Buttons_ButtonPress,
+                   &(*message_)[strings::msg_params],
+                   true);
+  } else if (!button_name_matches_module_type) {
+    LOG4CXX_WARN(logger_, "Request module type and button name mismatch!");
+    SendResponse(false,
+                 mobile_apis::Result::INVALID_DATA,
+                 "Request module type and button name mismatch!");
+  } else {
+    LOG4CXX_WARN(logger_, "Requested button is not exists in capabilities!");
+    SendResponse(false,
+                 mobile_apis::Result::UNSUPPORTED_RESOURCE,
+                 "Requested button is not exists in capabilities!");
+  }
+}
+
+AcquireResult::eType ButtonPressRequest::AcquireResource(
+    const app_mngr::commands::MessageSharedPtr& message) {
+  const std::string module_type =
+      (*message_)[app_mngr::strings::msg_params][message_params::kModuleType]
+          .asString();
+    app_mngr::ApplicationSharedPtr app =
+      application_manager_.application(CommandRequestImpl::connection_key());
+  return resource_allocation_manager_.AcquireResource(module_type, app->app_id());
+}
+
+bool ButtonPressRequest::IsResourceFree(const std::string& module_type) const {
+  return resource_allocation_manager_.IsResourceFree(module_type);
```

Comment:
@LitvinenkoIra this logic looks rather common. I suppose it should be moved to RCCommand

Reply:
@LuxoftAKutsan not all rc RPCs need to manage the resource